### PR TITLE
feat: bump jll version and add opt pass

### PR DESCRIPTION
**Pull request by avik-pal**

### CODE REVIEW

**Comment by avik-pal** [comment]
@wsmoses did something change in Enzyme in the new JLL? https://github.com/EnzymeAD/Reactant.jl/actions/runs/12405818372/job/34633303834?pr=400#step:9:488

**Comment by wsmoses** [comment]
Oh excellent. Yeah I added a better error message for us accidentally segfaulting sometimes due to asking enzyme to return the wrong number of args from an autodiff call

File: src/Interpreter.jl [hunk 1]
```diff
@@ -283,7 +283,7 @@ function overload_autodiff(
         end
     end
     for (i, act) in enumerate(activity)
-        if act == enzyme_out || (reverse && (act == enzyme_dup || act == enzyme_dupnoneed))
```
Comment (avik-pal) [inline src/Interpreter.jl:286]: I am not 100% sure if this is the right fix, but it fixes the tests

Comment (wsmoses) [reply src/Interpreter.jl:286]: yeah I think it is

**Pull request merged**